From cc139cdc3aa197784d32085af79fbf75b58f8bcc Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 7 Sep 2018 12:58:13 +0800
Subject: [PATCH 4563/5242] MLK-19484 drm/imx: dpu: crtc: Correct master
 extdst when plane_ed is slave

commit  b1d5a6389fc5ad8c6a9e4f4d08a06e1a8337975e from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch corrects master extdst when the relevant plane_ed of
the CRTC is slave.  This is code change only because there is no
real usecase supported by existing hardwares may trigger the issue.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dpu/dpu-crtc.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/imx/dpu/dpu-crtc.c b/drivers/gpu/drm/imx/dpu/dpu-crtc.c
index 9f06695..556e21f 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-crtc.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-crtc.c
@@ -101,8 +101,8 @@ static void dpu_crtc_atomic_enable(struct drm_crtc *crtc,
 			m_plane_ed = plane_ed;
 			s_plane_ed = aux_plane_ed;
 		} else {
-			m_plane_ed = plane_ed;
-			s_plane_ed = aux_plane_ed;
+			m_plane_ed = aux_plane_ed;
+			s_plane_ed = plane_ed;
 		}
 		extdst_pixengcfg_syncmode_master(m_plane_ed, true);
 		extdst_pixengcfg_syncmode_master(s_plane_ed, false);
-- 
1.7.9.5

